### PR TITLE
feat: Include a featured set of extensions

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Featured extensions will be promoted
It can also read an online catalog of all the extensions

- The featured set is stored with Podman Desktop
- The available set of extensions is stored online

This PR is just about the API to grab featured extensions (no display)

### What issues does this PR fix or reference?
#1901

### How to test this PR?

unit tests 

Change-Id: I76768894143a0d98c6184b73035d0591a4175087

### CODE REVIEW

**Comment by benoitf** [comment]
I've added more reviewers mostly for awareness

File: featured.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    {
+      "extensionId": "redhat.redhat-sandbox",
+      "displayName": "Developer Sandbox",
+      "shortDescription": "Free remote OpenShift sandbox environment for immediate access",
+      "categories": ["Kubernetes"],
```
Comment (jeffmaury) [inline featured.json:46]: Adding OpenShift as well ?

**Review by jeffmaury — COMMENTED** [review]
LGTM

Seems we pick the first version of the extension catalog so we will have to remember to always add first and I remember extension can set their minimum Podman Desktop version so is this reserved for future ?

File: featured.json (hunk 1 above)
Comment (benoitf) [reply featured.json:46]: Categories are predefined set. keywords/tags could be custom but categories are generic.

**Comment by benoitf** [comment]
> Seems we pick the first version of the extension catalog so we will have to remember to always add first and I remember extension can set their minimum Podman Desktop version so is this reserved for future ?

yes the JSON is how a REST API is serving content. it's latest first (filtering out the preview)

File: featured.json (hunk 1 above)
Comment (benoitf) [reply featured.json:46]: you can see that OpenShift is already added as a tag there: https://registry.podman-desktop.io/api/extensions.json

**Review by lstocchi — APPROVED** [review]
LGTM

The only change i would do is about the icons used as commented in the other PR https://github.com/containers/podman-desktop/pull/2355#pullrequestreview-1414737301

**Review by cdrage — APPROVED** [review]
Ran the unit tests (as requested). The code also LGTM! Yay including a list of featured extensions 🥳

**Pull request merged**